entity-service/src/main/kotlin/com/egm/stellio/entity/repository/Neo4jRepository.kt

```diff
@@ -581,6 +581,97 @@ class Neo4jRepository(
         }.flatten()
     }
 
+    fun getAttribute(): List<String> {
+        val query =
+            """
+                MATCH (a:Attribute)
+                OPTIONAL MATCH (a:Attribute)<-[:HAS_OBJECT|:HAS_VALUE]-(entity:Entity)
+                RETURN DISTINCT(labels(a))+collect(DISTINCT(a.name)) as attribute,
+                count(entity.location) as entityWithLocationCount
+            """.trimIndent()
+
+        val result = neo4jClient.query(query).fetch().all()
+        return result.map { rowResult ->
+            val entityWithLocationCount = (rowResult["entityWithLocationCount"] as Long).toInt()
+            var attributeList = (rowResult["attribute"] as List<String>)
+                .filter { it !in listOf("Attribute", "Relationship", "Property") }.toList()
+            if (entityWithLocationCount > 0) { attributeList += NGSILD_LOCATION_PROPERTY }
+            attributeList
+        }.flatten().sortedBy { it }
+    }
+
+    fun getAttributeDetails(): List<Map<String, Any>> {
+        val query =
```

Comment:
you'll have a lot of duplicate results with this query. for instance, an attribute will appear as many times as there are entities it is used in.